client/task-list/tasks/payments/RemotePayments/components/PaymentMethod.js

```diff
@@ -92,8 +92,8 @@ export const PaymentMethod = ( {
 		.map( ( pluginSlug ) => pluginNames[ pluginSlug ] )
 		.join( ' ' + __( 'and', 'woocommerce-admin' ) + ' ' );
 
-	const installStep =
-		plugins && plugins.length
+	const installStep = useMemo( () => {
```

Comment:
Curious why we should `useMemo` around the whole step versus using `React.memo` for just the `Plugin` component.  Is this a re-render issue with the plugin component or something else?

Reply:
There actually was excessive rendering down the entire chain due to redeclaring functions on every render, forcing every component that accepts those functions to rerender as well. 

I actually experimented with `React.memo`, but it didn't solve the issue. I just tried again as you described but I'm still seeing _way_ too many remounts for the `<Stepper />` component, which ends up remounting `<Plugins />` in turn:

![image](https://user-images.githubusercontent.com/444632/119174528-8ac64680-ba1d-11eb-9fc4-a9ea1afbc0cd.png)

That's with this code: 
```js
const MemoizedPlugins = memo( Plugins );

const installStep =
	plugins && plugins.length
		? {
				key: 'install',
				label: sprintf(
					/* translators: %s = one or more plugin names joined by "and" */
					__( 'Install %s', 'woocommerce-admin' ),
					pluginNamesString
				),
				content: (
					<MemoizedPlugins
						onComplete={ ( installedPlugins, response ) => {
							createNoticesFromResponse( response );
							recordEvent(
								'tasklist_payment_install_method',
								{
									plugins,
								}
							);
						} }
						onError={ ( errors, response ) =>
							createNoticesFromResponse( response )
						}
						autoInstall
						pluginSlugs={ plugins }
					/>
				),
				isComplete: ! pluginsToInstall.length,
			}
		: null;
```

I don't have a lot of experience with these memoization tools, but my suspicion was just that the `installStep` becomes a new object on every render, which in turn causes excessive remounts down the line. Let me know if I'm doing anything amiss above, but memoizing the `installStep` object and the `<DefaultStepper />` based on that seemed to do the trick.